cogs/birthday/birthday.py

```diff
@@ -416,6 +417,178 @@ async def deleteMemberBirthday(self, ctx: Context, member: discord.Member):
+            Birth month.
+        day: int
+            Birth day.
+        """
+        fnTitle = "Birthday - Set Self's Birthday"
+        headerBad = f":negative_squared_cross_mark: {bold(fnTitle)}"
+        headerGood = f":white_check_mark: {bold(fnTitle)}"
+
+        if not await self.config.guild(ctx.guild).get_attr(KEY_ALLOW_SELF_BDAY)():
+            await ctx.send(
+                f"{headerBad}\n"
+                "This function is not enabled. You cannot set your birthday. "
+                "Please let an administrator or a moderator know if you "
+                "believe this function should be enabled."
+            )
+            return
+
+        birthdayConfig = self.config.member(ctx.author)
+        if birthdayConfig:
+            birthMonth = birthdayConfig.get_attr(KEY_BDAY_MONTH)
+            birthDay = birthdayConfig.get_attr(KEY_BDAY_DAY)
+            if birthMonth and birthDay and await birthMonth() and await birthDay():
+                await ctx.send(
+                    f"{headerBad}\n"
+                    "Your birthday is already set. If you believe it is "
+                    "incorrect, please contact an admin or a moderator."
+                )
+                return
+
+        birthdayRoleId = await self.config.guild(ctx.guild).get_attr(KEY_BDAY_ROLE)()
+
+        if not birthdayRoleId:
+            await ctx.send(
+                f"{headerBad}\n"
+                "This server is not configured, please let a server "
+                "administrator or moderator know."
+            )
+            return
+
+        if not month and not day:
+            today = date.today()
+            month, day = today.month, today.day
+
+        try:
+            birthday = date(2020, month, day)
+            if birthdayConfig:
+                await birthdayConfig.get_attr(KEY_BDAY_MONTH).set(month)
+                await birthdayConfig.get_attr(KEY_BDAY_DAY).set(day)
+                await birthdayConfig.get_attr(KEY_ADDED_BEFORE).set(True)
+            else:
+                raise Exception(
+                    "Error while accessing member's birthday config. This should not happen!"
+                )
+        except ValueError:
+            await ctx.send(f"{headerBad}\n" "Please enter a valid birthday!")
+            return
+
+        birthdayStr = "{0:%B} {0:%d}".format(birthday)
+
+        sentReplyMsg: discord.Message = await ctx.send(
+            f"{headerGood}\n"
+            f"Successfully set your birthday to {bold(birthdayStr)}.\n"
+            f"{italics('(For your privacy, your birthday shown will be hidden shortly.)')}"
+        )
+
+        self.logger.info(
+            "%s#%s (%s) added their birthday as %s",
+            ctx.author.name,
+            ctx.author.discriminator,
+            ctx.author.id,
+            birthday.strftime("%B %d"),
+        )
+
+        await asyncio.sleep(5)
+        await sentReplyMsg.edit(content=(f"{headerGood}\n" "Successfully set your birthday."))
```

Comment:
The intention was that the other string is a line below the header but `black` reformatted since they are not long enough to stay on 2 separate lines of code.